### ISSUE TITLE
Missing header file for TSQLiteServer

### ISSUE THREAD

**Issue by benkrikler** [alcap-org/AlcapDAQ#127]
On develop if I do `make clean; make` I get compile errors from PulseCandidateFinder:

```
Making object: 'obj/PulseCandidateFinder.o'
src/TAP_generators/utils/PulseCandidateFinder.cpp:10:27: fatal error: TSQLiteServer.h: No such file or directory
 #include <TSQLiteServer.h>
                           ^
compilation terminated.
```

I checked and whilst the file in question exists in `root/sql/sqlite/inc`, this hasn't been copied into my `root-install/include` directory.  

I don't see any changes in the `install-root.sh` script though.  I can copy these files into `root-install/include`, but then this creates other issues.  

@AndrewEdmonds11 is it possible you changed the rootana makefile and / or install scripts without committing this?

**Comment by litchfld** [comment]
Sounds like root didn't build against SQLite? Hm.  What does `root-config --has-sqlite` say?

**Comment by AndrewEdmonds11** [comment]
@benkrikler, have you also installed sqlite with the install-sqlite3.sh script?

Also, I did change the makefile to add -lRSQLite to the ROOT libs so that it would compile.

**Comment by benkrikler** [comment]
I already ran the install-sqlite3.sh script, though this is a root problem anyway cause TSQLiteServer is a root class.

`root-config --has-sqlite` says `no` so the sqlite3 module in root was definitely not installed. I didn't run the install-root.sh script again because I saw there had been no changes to the install script.  Are we supposed to do this anyway? Does ROOT pick up on sqlite3 existing and compile things in automatically?

**Comment by AndrewEdmonds11** [comment]
So I had the exact same error and it was because I hadn't installed sqlite. Have you also re-run thisdaq.sh because I changed that as well

**Comment by AndrewEdmonds11** [comment]
Also, you can check that the option is "ON" in root-build/CMakeCache.txt

**Comment by benkrikler** [comment]
I had sourced thisdaq.sh again and just checked once more, but no change.

I can't check root-build because I removed the directory in a tidy up after root built.  Could this be the problem?  I'm going to set root recompiling just in case...

**Comment by benkrikler** [comment]
I've (finally) finished recompiling root and it's the same issue.  

In my CMakeCache.txt file, I have: 

``` cmake
sqlite:BOOL=OFF
```

If I change that and run again will it solve it?  Does this mean we need to change the root script?

**Comment by AndrewEdmonds11** [comment]
This is strange, so I haven't changed my root-install.sh script but in my CMakeCache.txt it says:

```
sqlite:BOOL=ON
```

(I'm not sure why I've got it and you haven't. Maybe mine found an install of sqlite somewhere already?)

If changing it solves it, then we can change the install script to include it.

**Comment by litchfld** [comment]
> I already ran the install-sqlite3.sh script, though this is a root problem anyway cause TSQLiteServer is a root class.

I'm not familiar with ROOT using CMake build system, but with the old configure system ROOT would hunt around for the libraries it needed for extensions and if it didn't find them it would quietly disable them. But to stop you compiling something you couldn't link it would not copy across the headers into the install area.  This sounds like the same behaviour. 

It used to be you could fix things by passing the location of the library in the configuration script, but i'm not sure how that happens now...

<time passes>

...OK.  I looked at my root install logs. 

It seems that my install picked up an existing SQLite package at `/usr/lib64/libsqlite3.so`, which kind of correlates with the fact I see no instructions to the root installer about our version.  The install we have it in AlcapDAQ is very pretty, but completely ornamental.  Presumably you don't have a system SQLite (which is of course why we wanted the install in the first place.)

So, the install-root script needs to be modified to look for the local copy of SQLite.  I'm still not sure how to do this, but it can't be too hard.

**Comment by litchfld** [comment]
> So, the install-root script needs to be modified to look for the local copy of SQLite. I'm still not sure how to do this, but it can't be too hard.

I'll take a look at tomorrow morning, unless someone gets there first.

PS: you (@benkrikler) can presumably check my hypothesis (that you have no system version of SQLite) by doing `which sqlite3` in a clean terminal.

**Comment by benkrikler** [comment]
Fixed it...

I had sqlite3:

```
$ which sqlite3
/bin/sqlite3
```

as well as the libsqlite3 c++ wrapper library.  I didn't have the devel package though, so I installed this via yum.  Previously I had tried reconfiguring cmake by going to the root-build directory and doing `ccmake .` which opens an ncurses interactive display.  When I did that however, it would not allow me to set sqlite to `on`.  Now, with the library installed, I was able to change this value.  I then did configure and generate which update the CMakeCache.txt file.   Then I re-ran `make; make install` and I now have sqlite in root:

```
$ root-config --has-sqlite
yes
```

and things now compile :smile:.

We do need to fix the install script though for long term use.  The root page on cmake doesn't mention much about sqlite unfortunately: http://root.cern.ch/drupal/content/building-root-cmake.
Looking at the cmake source files, I would guess that as well as setting `sqlite=on`, the following variables should be made to point to the alcap sqlite install:

```
SQLITE_INCLUDE_DIR
SQLITE_LIBRARIES
```

If we require a testing platform that doesn't have sqlite3 I can uninstall the devel packages I added to get it working and try things out here...

**Comment by benkrikler** [comment]
I have added in what I think are the right flags and make sure we test if sqlite is already compiled via the `install-sqlite.sh` script.

I checked this by removing the libsqlite3x packages through yum and running the install script up to where it executes make.  If I look in the CMakeCache.txt file sqlite is now on and there are a bunch more sqlite lines which all seem to point in the right place, so I think this is fixed.
